protosocket-connection/src/connection.rs

```diff
@@ -239,7 +239,7 @@ where
 
     pub fn read_inbound_messages_into_read_queue(&mut self) -> Result<bool, std::io::Error> {
         while self.receive_buffer_start_offset < self.receive_buffer_slice_end {
-            if 0 == self.inbound_messages.len() - self.inbound_messages.capacity() {
+            if self.inbound_messages.len() >= self.inbound_messages.capacity() {
```

Comment:
oh this existing code is just wrong. It's supposed to check for capacity... and capacity should be greater than len unless it's full. >= works, but == is intent-revealing (and I don't like `>` ;-))

```suggestion
            if self.inbound_messages.capacity() == self.inbound_messages.len()  {
```